src/query/ast/src/visitors/visitor_mut.rs

```diff
@@ -431,7 +431,11 @@ pub trait VisitorMut: Sized {
 
     fn visit_insert(&mut self, _insert: &mut InsertStmt) {}
     fn visit_replace(&mut self, _replace: &mut ReplaceStmt) {}
-    fn visit_merge_into(&mut self, _merge_into: &mut MergeIntoStmt) {}
+    fn visit_merge_into(&mut self, merge_into: &mut MergeIntoStmt) {
```

Comment:
`join_expr` and `merge_options` also need to visit

Reply:
The `Vistor` trait should also be changed to be consistent with `VisitorMut`

Reply:
we don't need to visit join_expr, because the target table and source is separate for now, it's not a join expr here. We just need to visit source and merge_options here.

Reply:
yes

Reply:
https://github.com/datafuselabs/databend/blob/main/src/query/ast/src/ast/expr.rs#L72
I think if we add these 28 exprs to our tests, we can cover all cases for the new changes. cc @BohuTANG @b41sh